docs/extending/template_components.rst

```diff
@@ -0,0 +1,167 @@
+Template components
```

Comment:
**Question** Just wondering why this new page for the docs is started in RST? As I understand we would like to migrate all documentation from RST to MD (with a few pages already started). It might be a helpful nudge in that direct to start all new documentation pages in MD directly.

Reply:
Mostly just inertia :-) In this case, there are a few function definitions and Intersphinx cross-links to the Django docs - looking up how to do those in MD (or whether I should revert to RST for those) would have been one extra thing to think about. Happy to switch if people think that would take things in a positive direction though!